x/auth/ante/unordered.go

```diff
@@ -76,3 +92,44 @@ func (d *UnorderedTxDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, _ bool, ne
 
 	return next(ctx, tx, false)
 }
+
+// txIdentifier returns a unique identifier for a transaction that is intended to be unordered.
+func txIdentifier(timeout uint64, tx sdk.Tx) ([32]byte, error) {
+	feetx := tx.(sdk.FeeTx)
```

Comment:
Should we exclude client simulations and just return an empty or predefined hash?

Reply:
if we look at charging gas for the sha256 call, then we should run this in simuluations as well. Ill leave it as is to avoid confusion

Reply:
Simulations come with zero costs for the caller. Therefore it would be good to avoid cpu intense operations to prevent DoS attacks. You can still charge gas for a correct gas value but bypass the hash calculation.
Simulations would always return a false negative on the duplicate TX check though but that would be the tradeoff for security.

Reply:
added